common/etl/etl.go

```diff
@@ -135,6 +58,11 @@ type LoadCommitHandler func(key []byte, isDone bool)
 
 type TransformArgs struct {
 	ExtractStartKey []byte
+	ExtractEndKey   []byte
+	Chunks          [][]byte
+	FixedBits       int
+	BufferType      int
```

Comment:
maybe we can just have a factory here? either `etl.NewAppendBuffer` or `etl.NewSortableBuffer`? both receive a size and return an `etl.Buffer`: `BufferFactory: func(size uint64) Buffer`

Reply:
Are you talking about this one https://github.com/ledgerwatch/turbo-geth/blob/252473d2701147c072249636b3eea51247dc2d8c/common/etl/buffers.go#L158 or something else?

Reply:
yeah, to get rid of it and get rid of buffer type enum and instead just put `BufferFactory` instead of `BufferType`.

Reply:
it is both less code and more flexible if we want to implement our custom buffer somewhere